test/datasets/test_iwslt2016.py

```diff
@@ -1,46 +1,128 @@
 
-    seed = 1
     mocked_data = defaultdict(lambda: defaultdict(list))
-    valid_set = "tst2013"
-    test_set = "tst2014"
-
-    files_for_split, _ = _generate_iwslt_files_for_lang_and_split(16, src, tgt, valid_set, test_set)
-    src_file = files_for_split[src][split]
-    tgt_file = files_for_split[tgt][split]
-    for file_name in (src_file, tgt_file):
-        txt_file = os.path.join(temp_dataset_dir, file_name)
-        with open(txt_file, "w") as f:
-            # Get file extension (i.e., the language) without the . prefix (.en -> en)
-            lang = os.path.splitext(file_name)[1][1:]
-            for i in range(5):
-                rand_string = " ".join(
-                    random.choice(string.ascii_letters) for i in range(seed)
-                )
-                dataset_line = f"{rand_string} {rand_string}\n"
-                # append line to correct dataset split
-                mocked_data[split][lang].append(dataset_line)
-                f.write(f'{rand_string} {rand_string}\n')
-                seed += 1
+
+    cleaned_file_names, uncleaned_file_names = _generate_iwslt_files_for_lang_and_split(16, src, tgt, valid_set, test_set)
+    uncleaned_src_file = uncleaned_file_names[src][split]
+    uncleaned_tgt_file = uncleaned_file_names[tgt][split]
+
+    cleaned_src_file = cleaned_file_names[src][split]
+    cleaned_tgt_file = cleaned_file_names[tgt][split]
+
+    for (unclean_file_name, clean_file_name) in [
+        (uncleaned_src_file, cleaned_src_file),
+        (uncleaned_tgt_file, cleaned_tgt_file)
+    ]:
+        # Get file extension (i.e., the language) without the . prefix (.en -> en)
+        lang = os.path.splitext(unclean_file_name)[1][1:]
+        expected_clean_filename = os.path.join(inner_temp_dataset_dir, clean_file_name)
+
+        # If we've already written a clean file, read it, so we don't generate
+        # new random strings. Otherwise generate new files and clean when read.
+        if os.path.exists(expected_clean_filename):
+            with open(expected_clean_filename, encoding="utf-8") as f:
+                mocked_data[split][lang] = f.readlines()
+        else:
+            out_file = os.path.join(inner_temp_dataset_dir, unclean_file_name)
+            with open(out_file, "w") as f:
+                mocked_data_for_split, file_contents = _generate_uncleaned_contents(split)
+                mocked_data[split][lang] = mocked_data_for_split
+                f.write(file_contents)
+
+    inner_compressed_dataset_path = os.path.join(
+        outer_temp_dataset_dir, f"{src}-{tgt}.tgz"
+    )
+
+    # create tar file from dataset folder
+    with tarfile.open(inner_compressed_dataset_path, "w:gz") as tar:
+        tar.add(inner_temp_dataset_dir, arcname=f"{src}-{tgt}")
+
+    outer_temp_dataset_path = os.path.join(
+        root_dir, "IWSLT2016", "2016-01.tgz"
+    )
+    with tarfile.open(outer_temp_dataset_path, "w:gz") as tar:
+        tar.add(outer_temp_dataset_dir, arcname="2016-01")
```

Comment:
Before adding `outer_temp_dataset_dir ` to the archive, we probably want to remove `inner_temp_dataset_dir` since the contents of this folder are already available in the `inner_compressed_dataset_path`, right?

Reply:
We could approach it that way -- my approach is to just re-read the cleaned files to generate the expected values. I don't know if there's a strong reason to prefer one vs. the other.